lib/Migration/Version3500Date20231005091430.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\Mail\Migration;
+
+use Closure;
+use JsonException;
+use OCA\Mail\AppInfo\Application;
+use OCP\DB\ISchemaWrapper;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\DB\Types;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+use Psr\Log\LoggerInterface;
+
+class Version3500Date20231005091430 extends SimpleMigrationStep {
+	/** @var IConfig */
+	protected $config;
+
+	/** @var IDBConnection */
+	protected $connection;
+
+	/** @var LoggerInterface */
+	protected $logger;
+
+	public function __construct(IConfig $config, IDBConnection $connection, LoggerInterface $logger) {
+		$this->config = $config;
+		$this->connection = $connection;
+		$this->logger = $logger;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		$schema = $schemaClosure();
+
+		if ($schema->hasTable('mail_provisionings')) {
```

Comment:
assume the table to exist but check if the columns do

this allows the migration to be executed more than once, e.g. for a rerun on error.

Reply:
Yes, that makes sense :)